homeassistant/components/simplisafe/config_flow.py

```diff
@@ -82,16 +82,31 @@ async def async_step_user(
             )
 
         errors = {}
-        session = aiohttp_client.async_get_clientsession(self.hass)
 
+        if len(auth_code := user_input[CONF_AUTH_CODE]) != 45:
+            # SimpliSafe authorization codes are 45 characters in length:
+            errors = {CONF_AUTH_CODE: "invalid_auth_code_length"}
+        elif auth_code.startswith("="):
+            # Ensure the user isn't including the "=" from the URL query param:
+            errors = {CONF_AUTH_CODE: "invalid_auth_code_start"}
```

Comment:
Why error in these cases? Alternatively, you could just strip it off?
Throwing the user an error, while you could solve the issue for them, isn't really friendly IMHO.

Reply:
> Why error in these cases?

The first error isn't fixable by us; if the user inputs something that's the wrong length, we can't do anything. Stripping off the `=` in the second case is a good idea. 👍🏻

Reply:
Might be worth adding this check first then, before checking length